spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -248,37 +229,76 @@ static void validateFeatureGeneratorConfig(Schema inputSchema, String outputColu
    *
-   * @param input      input Structured Record
+   * @param inputSchema input schema coming in from the previous stage
+   * @param key text field on which to perform text based feature generation
+   */
+  static void validateTextField(Schema inputSchema, String key) {
+    if (inputSchema.getField(key) == null) {
+      throw new IllegalArgumentException(String.format("Input field %s does not exist in the input schema %s.",
+                                                       key, inputSchema.toString()));
+    }
+    Schema schema = inputSchema.getField(key).getSchema();
+    Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+    if (type.equals(Schema.Type.ARRAY)) {
+      Schema componentSchema = schema.getComponentSchema();
+      type = componentSchema.isNullable() ? componentSchema.getNonNullable().getType() :
+        componentSchema.getType();
+    }
+    if (!(type.equals(Schema.Type.STRING) || type.equals(Schema.Type.ARRAY))) {
+      throw new IllegalArgumentException(String.format("Field to be transformed should be of type String or " +
+                                                         "Nullable String or Array of type String or Nullable " +
+                                                         "String . But was %s for field %s.", type, key));
+    }
+  }
+
+  /**
+   * Gets the input field for feature generation. If the field is an array, returns the field. Otherwise, splits the
+   * field based on the given pattern. Assumes the input field is an array of strings or a string.
+   *
+   * @param input input Structured Record
    * @param inputField field to use for feature generation
-   * @param pattern pattern to use for splitting the string
-   * @return
+   * @param splitter Splitter object to be used for splitting the input string
+   * @return text to be used for feature generation
    */
-  static List<String> getInputFieldValue(StructuredRecord input, String inputField, String pattern) {
+  static List<String> getInputFieldValue(StructuredRecord input, String inputField, Splitter splitter) {
     List<String> text = new ArrayList<>();
     Schema schema = input.getSchema().getField(inputField).getSchema();
-    if (!schema.isNullable() && input.get(inputField) == null) {
-      throw new IllegalArgumentException(String.format("Null value found for non-nullable field %s", inputField));
-    }
     Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
     if (type.equals(Schema.Type.ARRAY)) {
       schema = schema.getComponentSchema();
-      if (!schema.isNullable() && ((ArrayList) input.get(inputField)).isEmpty()) {
-        throw new IllegalArgumentException(String.format("Null value found for non-nullable field %s", inputField));
-      }
       type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
       if (type.equals(Schema.Type.STRING)) {
```

Comment:
it's easier to read if you do:

```
if (type != Schema.Type.String)) {
  throw new ...
}

//logic here
```
as it removes one layer of indentation. In any case, I'm not sure if you even need the check. Isn't this validated at configure time?